tests/dnn/dnn-layer-batchnormalization-test.c

```diff
@@ -0,0 +1,132 @@
+    /*
+    import tensorflow as tf
+    import numpy as np
+
+    x = tf.placeholder(tf.float32, shape=[1, None, None, 3])
+    in_data = np.random.rand(1, 5, 4, 3)
+
+    offset = 0.1
+    scale = 1.0
+    var_eps = 0.0001
+    mean = tf.constant([0.1, 0.15, 0.14], dtype=np.float32)
+    var = tf.constant([0.12, 0.21, 0.24], dtype=np.float32)
+
+    y = tf.nn.batch_normalization(x, mean, var, offset, scale, var_eps)
+
+    sess = tf.Session()
+    sess.run(tf.global_variables_initializer())
+    output = sess.run(y, feed_dict={x: in_data})
+
+    print('input_data:')
+    print(in_data.shape)
+    print(list(in_data.flatten()))
+
+    print('output:')
+    print(output.shape)
+    print(list(output.flatten()))
+    */
+    DnnLayerBatchnormalizationParams params;
+    DnnOperand operands[2];
+    int32_t input_indexes[1];
+    float input_data[1*5*4*3] = {
+        0.03592029475149139, 0.8177031019998039, 0.7439828967882774, 0.12351433689912827, 0.10151796406770697, 0.3746884445121409, 0.17609107818009584,
+        0.9180125678699762, 0.413786266728411, 0.15388845485290492, 0.6527147025219426, 0.8921708075708997, 0.9315511108099018, 0.6089771636865586,
+        0.9096167333314585, 0.35798486932034357, 0.242417367935445, 0.3354539420380003, 0.5885835336919439, 0.2330763092626813, 0.19528398816193004,
+        0.3537750057949528, 0.8908506440029637, 0.6745876262153178, 0.7975390896917102, 0.42115552951085633, 0.026252384244236926, 0.5987492499419249,
+        0.2969445642379439, 0.6612326239920433, 0.5186612970859783, 0.9154125186721356, 0.11647732132450173, 0.32631441601002287, 0.5222573083721183,
+        0.47294093689301375, 0.009744493333811666, 0.034835869186598645, 0.14643674017261454, 0.06114675456320906, 0.07678534832197992, 0.1968478741676435,
+        0.3126360178524745, 0.6361975996395828, 0.18012602596211258, 0.8609077898378013, 0.8446431696067955, 0.7497824163136522, 0.8651808467465752,
+        0.922506696896085, 0.6057714540829449, 0.6226843061111148, 0.3894522432026163, 0.15585932050469298, 0.5738772930525835, 0.4978427887293949,
+        0.6015976578669766, 0.9169836327963268, 0.6957180046975255, 0.49003527036757355
+    };
+    float expected_output[1*5*4*3] = {
+        -0.084905155, 1.5567008, 1.3326182, 0.16785178, -0.005771324, 0.578956, 0.31956452, 1.7755418, 0.65874743, 0.25549775, 1.1967523,
+        1.6350424, 2.4994817, 1.1013318, 1.6706462, 0.844428, 0.3016232, 0.4988855, 1.5098319, 0.2812442, 0.21282446, 0.8322802, 1.7162839,
+        1.190995, 2.1127834, 0.6915691, -0.13213801, 1.5391655, 0.42058298, 1.1637402, 1.3080678, 1.7698696, 0.051994503, 0.7530414,
+        0.9121387, 0.7794713, -0.16043672, -0.15124893, 0.11313617, -0.01211296, -0.059729457, 0.21601605, 0.7135717, 1.1607176, 0.18188983,
+        2.295637, 1.6154748, 1.344454, 2.307967, 1.7853465, 1.050554, 1.6082313, 0.62240326, 0.13236594, 1.4673963, 0.85887456,
+        1.042036, 2.4574463, 1.2905709, 0.8143577
```

Comment:
is it possible to calculate expected_output easily with c code? If yes, don't use the pre-calculated values.

Reply:
Yes, it is possible to use c code to get expected_output.